streaming/base/shuffle/py1e.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Shuffling algorithm that shuffles intra-shard in one place.
```

Comment:
this is the description for py1s, need to tweak it for py1e

Reply:
done!